### DISCUSSION TITLE
unocss 在 naive ui 中的使用

### DISCUSSION THREAD

**Discussion by wendraw** [tusen-ai/naive-ui#4800] [Q&A]
naive ui 源码与 unocss 一起使用的时候打包会卡住 vite dev 的进程是为什么？

**Comment by 07akioni** [comment]
这个不太好说，naive 自身只是一堆 js 代码，不会参与任何编译期的流程，建议还是检查一下完整的配置

**Reply by wendraw** [reply]
好的，这个问题我找到了答案，是 unocss 的 [transformerAttributifyJsx](https://unocss.dev/transformers/attributify-jsx) 在编译的时候会有问题，具体原因没有深究。配上 exclude 过滤掉 naive-ui 的源码目前能正常工作。

还遇到另一个问题，想请教一下，用 [@vitejs/plugin-vue-jsx](https://github.com/vitejs/vite-plugin-vue/tree/main/packages/plugin-vue-jsx) 编译 naive-ui 源码，在使用 `Popover` 相关组件时会报错，其内容也不会被渲染出来。

使用 `esbuild: {  jsxFactory: 'h',  jsx: 'transform', jsxFragment: 'Fragment' }` 配置来编辑代码就没问题，我有个疑问是在生产环境，vite 打包 esbuild 不会介入吧。这样会不会有问题？虽然目前使用 `view build` 编译代码能正常运行。

有相关的技术交流群吗？这里提问的效率有点低。

**Reply by 07akioni** [reply]
看 readme，有

**Reply by 07akioni** [reply]
naive 是标准 tsx，需要用通用工具编译

**Reply by 07akioni** [reply]
naive 不发布源码，发布出去的都是 js，不存在你说的问题
